### PR TITLE
Add ecdsa support

### PR DESCRIPTION
**Pull request by barkbay**
This PR adds support for the ECDSA signature algorithm.

This enhancement is compatible with the support of a custom CA for [HTTP](https://github.com/elastic/cloud-on-k8s/pull/4575) and [transport](https://github.com/elastic/cloud-on-k8s/pull/4053) layer.

I'll try to add an e2e test but I'm opening this PR to allow a first review of the design/implementation, which is mostly about replacing `*rsa.PrivateKey` with the `crypto.Signer` interface.

Fix https://github.com/elastic/cloud-on-k8s/issues/4581

### CODE REVIEW

**Comment by barkbay** [comment]
jenkins test this please

**Review by pebrc — APPROVED** [review]
The change looks good. 👍 on an e2e test. I ran into the issue that the first curve I tried,  secp256k1, was not supported. I believe Elasticsearch claims support for all curves mentioned  in rfc5480 and rfc5915 (secp256k1 is not in those rfcs). But I could not find information on which curves are supported in Go.

**Comment by barkbay** [comment]
jenkins test this please

File: pkg/controller/common/certificates/pem.go [hunk 1]
```diff
@@ -108,15 +143,54 @@ func GetPrimaryCertificate(pemBytes []byte) (*x509.Certificate, error) {
...
-func PrivateMatchesPublicKey(publicKey interface{}, privateKey rsa.PrivateKey) bool {
-	pubKey, ok := publicKey.(*rsa.PublicKey)
-	if !ok {
-		log.Error(errors.New("Public key is not an RSA public key"), "")
+func PrivateMatchesPublicKey(publicKey crypto.PublicKey, privateKey crypto.Signer) bool {
+	switch k := publicKey.(type) {
+	case *rsa.PublicKey:
+		return k.Equal(privateKey.Public())
+	case *ecdsa.PublicKey:
+		return k.Equal(privateKey.Public())
```
Comment (thbkrkr) [inline pkg/controller/common/certificates/pem.go:151]: Could be:
```suggestion
	case *rsa.PublicKey, *ecdsa.PublicKey:
		return k.Equal(privateKey.Public())
```

File: pkg/controller/common/certificates/pem.go [hunk 2]
```diff
@@ -108,15 +143,54 @@ func GetPrimaryCertificate(pemBytes []byte) (*x509.Certificate, error) {
...
+	default:
+		log.Error(fmt.Errorf("unsupported public key type: %T", publicKey), "")
 		return false
 	}
-	// check that public and private keys share the same modulus and exponent
-	if pubKey.N.Cmp(privateKey.N) != 0 || pubKey.E != privateKey.E {
-		return false
+}
+
+// GetCompatiblePrivateKey returns a PEM encoded private key iff the ca and the key have the same underlying type.
```
Comment (thbkrkr) [inline pkg/controller/common/certificates/pem.go:158]:
```suggestion
// GetCompatiblePrivateKey returns a PEM encoded private key iff the CA and the key have the same underlying type.
```

**Review by thbkrkr — APPROVED** [review]
Nice work, LGTM.

**Pull request merged**